expect_test.go

```diff
@@ -821,6 +821,51 @@ func TestSpawnGeneric(t *testing.T) {
 	}
 }
 
+// TestSendTimeout tests that Send command can fail on timeout.
+func TestSendTimeout(t *testing.T) {
+	t.Log("Running test: TestSendTimeout")
+	rIn, wIn := io.Pipe()
+	rOut, wOut := io.Pipe()
+	waitCh := make(chan error)
+	outCh := make(chan string)
+	defer close(outCh)
+
+	go fakeCli(cliMap, rIn, wOut)
+	exp, r, err := SpawnGeneric(
+		&GenOptions{
+			In:    wIn,
+			Out:   rOut,
+			Wait:  func() error { return <-waitCh },
+			Close: func() error { return nil },
+			Check: func() bool { return true },
+		}, -1, SendTimeout(10*time.Second))
+	if err != nil {
+		t.Errorf("test: TestSendTimeout, SpawnGeneric failed: %v", err)
+	}
+
+	err = wIn.Close()
+	if err != nil {
+		t.Errorf("test: TestSendTimeout, wIn.Close() failed: %v", err)
+	}
+
+	err = exp.Send("test" + "\n")
+	if err != nil {
+		t.Errorf("test: TestSendTimeout, Send command failed: %v", err)
+	}
+
+	err = exp.Send("test" + "\n")
+	if err == nil {
+		t.Error("test: TestSendTimeout, Send command must failed with the timeout")
+	}
+
+	waitCh <- nil
+	exp.Close()
+	wOut.Close()
```

Comment:
Could this be deferred?

Reply:
I think I can not because we get the error from the channel in the end `<- r` if I will call defer it will close the expecter only after this expression, and test will stuck forewer on `<- r`